### PR TITLE
Add wc client api docu

### LINKED ISSUE #3772: Add wc client api docu
Currently we don't have a webcomponent client api documentation like https://docs.luigi-project.io/docs/luigi-client-api
tbd

### PR DESCRIPTION
**Pull request by amilewskaa**
**Description**

Changes proposed in this pull request:

- add label-success to documentation
- update Luigi Client API page with labels

**Related issue(s)**
Resolves #3772

### CODE REVIEW

File: website/docs/src/markdown-conversion/unified-plugins/rehype-label.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { visit } from 'unist-util-visit';
```
Comment (ndricimrr) [inline website/docs/src/markdown-conversion/unified-plugins/rehype-label.js:1]: Could you document at the top in short: What this file does and what its used for very shortly. Had to review one of these, but was written 4 years ago. Makes it easier when documented

File: website/docs/src/markdown-conversion/unified-plugins/rehype-label.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+export default function addLabelTag() {
```
Comment (ndricimrr) [inline website/docs/src/markdown-conversion/unified-plugins/rehype-label.js:3]: can you add a JSDOC on this exported function header

File: website/docs/src/markdown-conversion/unified-plugins/rehype-label.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+      if (match) {
+        const className = match[1];
+        const labelText = match[2];
+        transformToLabel(node, index, parent, className, labelText);
+      }
+    });
+  };
+
+  function transformToLabel(node, index, parent, className, labelText) {
```
Comment (ndricimrr) [inline website/docs/src/markdown-conversion/unified-plugins/rehype-label.js:17]: jsdoc here as well. 

would be helpful to also visualise input/outputs what the HTML looks like with a very basic example too.

**Review by ndricimrr — COMMENTED** [review]
small comments

File: website/docs/src/markdown-conversion/unified-plugins/rehype-label.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ * @returns {function} Transformer function for the Markdown AST.
+ *
+ * @example
+ * // Input:
+ * // <!-- label-info: This is an info label -->
+ * //
+ * // Output:
+ * // <div class="label-wrapper">
+ * //   <span class="label-info">This is an info label</span>
+ * // </div>
```
Comment (ndricimrr) [inline website/docs/src/markdown-conversion/unified-plugins/rehype-label.js:17]:
```suggestion
 *    Input:
 *    <!-- label-info: This is an info label -->
 *   
 *    Output:
 *    <div class="label-wrapper">
 *      <span class="label-info">This is an info label</span>
 *    </div>
```

File: website/docs/src/markdown-conversion/unified-plugins/rehype-label.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+   * //   properties: { className: 'label-wrapper' },
+   * //   children: [
+   * //     {
+   * //       type: 'element',
+   * //       tagName: 'span',
+   * //       properties: { className: 'label-info' },
+   * //       children: [{ type: 'text', value: 'This is an info label' }]
+   * //     }
+   * //   ]
+   * // }
```
Comment (ndricimrr) [inline website/docs/src/markdown-conversion/unified-plugins/rehype-label.js:59]:
```suggestion
   *    Input Node:
   *    { type: 'comment', value: ' label-info: This is an info label ' }
   *   
   *    Output Node:
   *    {
   *      type: 'element',
   *      tagName: 'div',
   *      properties: { className: 'label-wrapper' },
   *      children: [
   *        {
   *          type: 'element',
   *          tagName: 'span',
   *          properties: { className: 'label-info' },
   *          children: [{ type: 'text', value: 'This is an info label' }]
   *        }
   *      ]
   *    }
```

**Review by ndricimrr — COMMENTED** [review]
i think the // can be removed. The * already enough

**Comment by JohannesDoberer** [comment]
The documentation in [docs/luigi-client-api.md](https://github.com/SAP/luigi/pull/3825/files#diff-4ff8ced4162e9cb3719d7544de4ec0ff185efbc3d3d655b610c857bb224e5f26) is generated from the jsdocs in luigi client src files (lifecycleManager.js, linkManager.js, uxManager.js etc).
You need to add the label `* <!-- label-success: Web Component API  -->` to the jsdocs.
E.g.: take a look at this [example](https://github.com/SAP/luigi/blob/main/client/src/lifecycleManager.js#L425)

To generate the docu you need to run [npm run docu](https://github.com/SAP/luigi/blob/main/scripts/package.json#L5) in `/scripts` folder.
After that you can run the docu webside like is described [here](https://github.com/SAP/luigi/blob/main/website/docs/README.md)

**Review by JohannesDoberer — APPROVED** [review]
LGTM 👍 !

**Pull request merged**